types/downloads.go

```diff
@@ -26,6 +26,16 @@ const (
 	DownloadTaskStatusDownloading = "downloading"
 )
 
+type downloadTaskError string
+
+const (
+	DownloadTaskErrorNone             downloadTaskError = "none"          // No error
+	DownloadTaskErrorHashDownloadDisk downloadTaskError = "hash_download" // Error downloading the hash file
+	DownloadTaskError4XX              downloadTaskError = "http_4xx"      // Error 4xx
+	DownloadTaskErrorNotFound         downloadTaskError = "not_found"     // Error 404
+	// TODO: Add more error codes
```

Comment:
What's with this todo?

Are the error code strings not documented and you've been reverse engineering them ? (If so kuddos)

Reply:
You are right, I added all the documented error code, and put my retro-engineered error codes in a dedicated section.

Wdyt about this new list?

Reply:
Sounds good to me, thanks for doin the heavy work, I know it's can be painful. Would you mind updating the TODO to mention that it's undocumented and that those errors were manually gathered ? (something of sorts, the wording is not super important)

Reply:
Perfectomundo 👌